File: app/src/main/java/com/univreview/view/contract/UploadReviewContract.java
package com.univreview.view.contract;

import android.content.Context;

import com.univreview.model.Review;

/**
 * Created by DavidHa on 2017. 7. 20..
 */

public interface UploadReviewContract extends BaseContract<UploadReviewContract.View> {
    interface View {
        void showProgress();
        void dismissProgress();
        void showRecommendRvDialog();
        String getSubjectName();
        String getProfessorName();
        void showAlertDialog();
        void setReviewExist(boolean isExist);
    }

    Review getReview();
    void setContext(Context context);
    void registerReview();
    void checkReviewExist();

}

File: app/src/main/java/com/univreview/view/contract/UploadReviewContract.kt
package com.univreview.view.contract

import android.content.Intent
import com.univreview.model.Review

/**
 * Created by DavidHa on 2017. 8. 6..
 */
interface UploadReviewContract {
    interface View : BaseView {
        fun showAlertDialog()
        fun showSimpleMsgDialog(msg: String)
        fun showRecommendRvDialog(review: Review)
        fun setSubjectTxt(str: String?)
        fun setProfessorTxt(str: String?)
    }

    fun onActivityResult(data: Intent)
    fun registerReview()
    fun checkReviewExist()
}